cloudsmith/resource_deny_rule.go

```diff
@@ -0,0 +1,198 @@
+		Name:               nullableString(d, "name"),
+		Enabled:            optionalBool(d, "enabled"),
+		Description:        nullableString(d, "description"),
+		PackageQueryString: nullableString(d, "package_query"),
+	})
+	denyRule, _, err := pc.APIClient.OrgsApi.OrgsDenyPolicyPartialUpdateExecute(req)
+	if err != nil {
+		return err
+	}
+	d.SetId(denyRule.GetSlugPerm())
+	checkerFunc := func() error {
+		// this is somewhat of a hack until we have a better way to poll for a
+		// deny rule being updated
+		time.Sleep(time.Second * 5)
+		return nil
+	}
+	if err := waiter(checkerFunc, defaultUpdateTimeout, defaultUpdateInterval); err != nil {
+		return fmt.Errorf("error waiting for deny rule (%s) to be updated: %w", d.Id(), err)
+	}
+	return denyRuleRead(d, m)
+}
+
+func denyRuleDelete(d *schema.ResourceData, m interface{}) error {
+	pc := m.(*providerConfig)
+
+	namespace := requiredString(d, "namespace")
+
+	req := pc.APIClient.OrgsApi.OrgsDenyPolicyDelete(pc.Auth, namespace, d.Id())
+	_, err := pc.APIClient.OrgsApi.OrgsDenyPolicyDeleteExecute(req)
+	if err != nil {
+		return err
+	}
+
+	checkerFunc := func() error {
+		req := pc.APIClient.OrgsApi.OrgsDenyPolicyRead(pc.Auth, namespace, d.Id())
+		if _, resp, err := pc.APIClient.OrgsApi.OrgsDenyPolicyReadExecute(req); err != nil {
+			if is404(resp) {
+				return nil
+			}
+			return err
+		}
+		return errKeepWaiting
+	}
+
+	if err := waiter(checkerFunc, defaultDeletionTimeout, defaultDeletionInterval); err != nil {
+		return fmt.Errorf("error waiting for deny rule (%s) to be deleted: %w", d.Id(), err)
+	}
+	return nil
+}
+
+//nolint:funlen
+func denyRule() *schema.Resource {
+	return &schema.Resource{
+		Create: denyRuleCreate,
+		Read:   denyRuleRead,
+		Update: denyRuleUpdate,
+		Delete: denyRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: denyRuleImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Description:  "A descriptive name for the deny rule.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"action": {
+				Type:         schema.TypeString,
+				Description:  "A descriptive name for the deny rule.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"slug_perm": {
+				Type:         schema.TypeString,
+				Description:  "The slug of the deny rule.",
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Description:  "Description of the rule.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"package_query": {
+				Type:         schema.TypeString,
+				Description:  "The query to match the packages to be blocked.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Is the rule enabled?.",
+				Optional:    true,
```

Comment:
should `enabled` maybe be required and we can set default value?

Reply:
If it is required, has to be specified by the user in the resource. e.g.:
```
resource "cloudsmith_package_deny_policy" "test-2" {
  name = "test-me-tf-2-2"
  description = "test"
  enabled = true
  package_query = "bananas"
  namespace = "cloudsmith-test"
}
```
So I think we want it to be optional with default `true` so we can do:
```
resource "cloudsmith_package_deny_policy" "test-2" {
  name = "test-me-tf-2-2"
  description = "test"
  package_query = "bananas"
  namespace = "cloudsmith-test"
}
```
And defaults to `true`. Does it make sense?

Reply:
Makes sense